### PR TITLE
Move all of gp_sparse_vector staff to sparse_vector schema

### PR DESCRIPTION
**Pull request by HaozhouWang**
Reason: the gp_sparse_vector has many functions and operators
	are the same as GPDB/PG built-in functions and operators.

	Currently, the extension will create its functions and
	operators in the public schema and will override the built-in
	functions and operators. As the behavior between extension
	functions and operators are different with built-in
	functions and operators. Hence, it will cause some issues.

In this PR, the gp_sparse_vector's functions and operators will be
put into a separated schema to split extension and built-in.
All tests are also updated.

P.S. This PR is for 6X_STABLE only

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [x] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: gpcontrib/gp_sparse_vector/gp_sparse_vector--1.0.0.sql [hunk 1]
```diff
@@ -369,3 +373,5 @@ OPERATOR        3       == ,
 OPERATOR        4       >= ,
 OPERATOR        5       > ,
 FUNCTION        1       svec_l2_cmp(svec, svec);
+
+SET search_path TO DEFAULT;
```
Comment (zhangh43) [inline gpcontrib/gp_sparse_vector/gp_sparse_vector--1.0.0.sql:377]: should we reset search_path here?

File: gpcontrib/gp_sparse_vector/sql/gp_svec.sql [hunk 2]
```diff
@@ -80,4 +82,4 @@ SELECT array_agg(a) FROM (SELECT trunc(7) a,generate_series(1,100000) ORDER BY a
 SELECT vec_median('{9960,9926,10053,9993,10080,10050,9938,9941,10030,10029}:{1,9,8,7,6,5,4,3,2,0}'::svec);
 SELECT vec_median('{9960,9926,10053,9993,10080,10050,9938,9941,10030,10029}:{1,9,8,7,6,5,4,3,2,0}'::svec::float8[]);
 
-DROP EXTENSION gp_sparse_vector;
```
Comment (zhangh43) [inline gpcontrib/gp_sparse_vector/sql/gp_svec.sql:83]: reset search_path here?

**Comment by zhangh43** [comment]
> Currently, the extension will create its functions and operators in the public schema and will override the built-in
functions and operators

How do we handle the existing user? The binary upgrade will not modify the existing extension. How about adding a new version 1.0.1 and following the extension upgrade logic?

File: gpcontrib/gp_sparse_vector/gp_sparse_vector--1.0.0.sql (hunk 1 above)
Comment (HaozhouWang) [reply gpcontrib/gp_sparse_vector/gp_sparse_vector--1.0.0.sql:377]:
```
SET search_path TO DEFAULT;
```
is the reset search_path

File: gpcontrib/gp_sparse_vector/sql/gp_svec.sql (hunk 2 above)
Comment (HaozhouWang) [reply gpcontrib/gp_sparse_vector/sql/gp_svec.sql:83]: will fix this.

**Comment by HaozhouWang** [comment]
Good idea, I will update it.

File: gpcontrib/gp_sparse_vector/sql/gp_svec.sql [hunk 3]
```diff
@@ -80,4 +82,5 @@ SELECT array_agg(a) FROM (SELECT trunc(7) a,generate_series(1,100000) ORDER BY a
...
\ No newline at end of file
+DROP EXTENSION gp_sparse_vector;
+SET search_path TO DEFAULT;
```
Comment (zhangh43) [inline gpcontrib/gp_sparse_vector/sql/gp_svec.sql:86]: the answer file seems not updated.

File: gpcontrib/gp_sparse_vector/test_output [hunk 4]
```diff
@@ -1,4 +1,5 @@
 -- \set FETCH_COUNT 100
+SET search_path TO sparse_vector;
```
Comment (zhangh43) [inline gpcontrib/gp_sparse_vector/test_output:2]: do we need reset search_path here?

File: gpcontrib/gp_sparse_vector/gp_sparse_vector--1.0.0--1.0.1.sql [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+ALTER OPERATOR + (svec, float8[])
+	SET SCHEMA sparse_vector;
+
+ALTER OPERATOR * (svec, float8[])
+	SET SCHEMA sparse_vector;
+
+ALTER OPERATOR / (svec, float8[])
+	SET SCHEMA sparse_vector;
+
+-- CAST no schema issue
```
Comment (zhangh43) [inline gpcontrib/gp_sparse_vector/gp_sparse_vector--1.0.0--1.0.1.sql:164]: comment `CAST no schema issue` is too simple, please share more context here including: detail of schema issue and cast doesn't not support alter schema(better to copy Tomlane's comment as well to help others get the context).

**Review by zhangh43 — COMMENTED** [review]
why do you rename 1.0.0 to 1.0.1? I think we should keep 1.0.0 and just add file 1.0.0->1.0.1. Please help to verity the following two cases:
1. new user runs `create extension gp_sparse_vector` should succeed and version should be 1.0.1
2. existing user with 1.0.0 installed, runs alter extension should succeed, version should be 1.0.1

**Comment by zhangh43** [comment]
@HaozhouWang Current patch require the existing user to rewrite their scripts(add schema prefix) which includes gp_sparse_vector.
Did you try `create extension with schema` syntax to fix the schema issue?

**Comment by lisakowen** [comment]
i am following the progress of this PR so that i can update the docs appropriately.  when you merge, can you please summarize what an existing user of the gp_sparse_vector module needs to do when they upgrade to a greenplum release that includes this change?  (i.e. do they need to drop/recreate the gp_sparse_vector extension?  alter extension gp_sparse_vector update?  prepend `sparse_vector.` to gp_sparse_vector functions referenced in any scripts?  other?).  thanks.

**Comment by HaozhouWang** [comment]
create extension with schema seems not work in GPDB 6

File: gpcontrib/gp_sparse_vector/test_output (hunk 4 above)
Comment (HaozhouWang) [reply gpcontrib/gp_sparse_vector/test_output:2]: Thanks fixed.

File: gpcontrib/gp_sparse_vector/sql/gp_svec.sql (hunk 3 above)
Comment (HaozhouWang) [reply gpcontrib/gp_sparse_vector/sql/gp_svec.sql:86]: fixed

File: gpcontrib/gp_sparse_vector/expected/gp_svec.out [hunk 6]
```diff
@@ -14,20 +16,21 @@ SELECT a,b::float8[] cross_product_equals FROM (SELECT a,b FROM test) foo WHERE
...
- {2,2,2,1,1,1}
- {2,1,2,1,1,1}
  {0,0,2,0,0,0}
  {2,2,2,1,1,1}
- {4,3,2,2,2,2}
+ {2,2,2,1,1,1}
  {4,4,2,2,2,2}
+ {2,1,2,1,1,1}
+ {4,3,2,2,2,2}
+ {4,2,2,2,2,2}
```
Comment (zhangh43) [inline gpcontrib/gp_sparse_vector/expected/gp_svec.out:33]: should we add order by to make the output order stable?

**Review by zhangh43 — APPROVED** [review]
Others looks good to me

File: gpcontrib/gp_sparse_vector/expected/gp_svec.out (hunk 6 above)
Comment (HaozhouWang) [reply gpcontrib/gp_sparse_vector/expected/gp_svec.out:30]: fixed

**Pull request merged**